modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -0,0 +1,610 @@
+        )
+    {
+        int ninputs = inputs.size();
+
+        std::vector<const char*> v_inp;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp), [] (const auto& m) { return m.template ptr<const char>(); });
+        const char** inp = v_inp.data();
+
+        std::vector<int> v_inp_dims;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_dims), [] (const auto& m) { return m.dims; });
+        const int* inp_ndims = v_inp_dims.data();
+
+        std::vector<const int*> v_inp_shape;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_shape), [] (const auto& m) { return m.size.p; });
+        const int** inp_shape = v_inp_shape.data();
+
+        std::vector<const size_t*> v_inp_step;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_step), [] (const auto& m) { return m.step.p; });
+        const size_t** inp_step = v_inp_step.data();
+
+        char* out = outputs[0].ptr<char>();
+
+        int out_ndims = outputs[0].dims;
+        const int* out_shape = outputs[0].size.p;
+        const size_t* out_step = outputs[0].step.p;
+
+        int i, max_ndims = out_ndims > 2 ? out_ndims : 2;
+        for(i = 0; i < ninputs; i++)
+            max_ndims = max_ndims > inp_ndims[i] ? max_ndims : inp_ndims[i];
+        size_t* buf = (size_t*)malloc((ninputs+1)*
+            (max_ndims*(sizeof(int) + sizeof(size_t))
+            + sizeof(void*)*5   // 5 arrays with pointers with shape [ninputs][maxdims]
+            + sizeof(int)       // array of dims with shape [ninputs]
+            + sizeof(size_t))); // array of element sizes with shape [ninputs]
+
+        int** orig_shapes = (int**)buf;
+        int** shapes = orig_shapes + ninputs + 1;
+        size_t** orig_steps = (size_t**)(shapes + ninputs + 1);
+        size_t** steps = orig_steps + ninputs + 1;
+        char** ptrs = (char**)(steps + ninputs + 1);
+
+        size_t* step_buf = (size_t*)(ptrs + ninputs + 1);
+        int* shape_buf = (int*)(step_buf + (ninputs + 1)*max_ndims);
+        int* all_ndims = shape_buf + (ninputs + 1)*max_ndims;
+        size_t* all_type_sizes = (size_t*)(all_ndims + ninputs + 1);
+
+        for(i = 0; i <= ninputs; i++) {
+            all_ndims[i] = i == 0 ? out_ndims : inp_ndims[i-1];
+            all_type_sizes[i] = sizeof(T);
+            orig_shapes[i] = (int*)(i == 0 ? out_shape : inp_shape ? inp_shape[i-1] : 0);
+            orig_steps[i] = (size_t*)(i == 0 ? out_step : inp_step ? inp_step[i-1] : 0);
+            shapes[i] = shape_buf + max_ndims*i;
+            steps[i] = step_buf + max_ndims*i;
+        }
+
+        if (!prepare_for_broadcast_op(ninputs + 1, max_ndims, all_type_sizes,
+                                      all_ndims, (const int**)orig_shapes,
+                                      (const size_t**)orig_steps,
+                                      shapes, steps))
+            return;
```

Comment:
RAII should be used for buf, memory leak here

Reply:
Does AutoBuffer solve this problem like what I did?